session.go

```diff
@@ -779,24 +789,12 @@ func (s *sessionImpl) Listen(ctx context.Context, cfg config.Tunnel) (Tunnel, er
 
 	if err != nil {
 		return nil, errListen{err}
+	} else {
+		return &tunnelImpl{
+			Sess:   s,
+			Tunnel: tunnel,
+		}, nil
 	}
-
-	t := &tunnelImpl{
-		Sess:   s,
-		Tunnel: tunnel,
-	}
-
-	if httpServerCfg, ok := cfg.(interface {
-		HTTPServer() *http.Server
-	}); ok {
-		if srv := httpServerCfg.HTTPServer(); srv != nil {
-			go func() {
-				_ = srv.Serve(t)
-			}()
-		}
-	}
-
-	return t, nil
```

Comment:
Hmm... not sure how I feel about actively *breaking* this behavior, even if the methods are deprecated. What if we went ahead and created our waitgroup here and added it to the `tunnelImpl`, and then if `ListenAndServe` is called afterward and we find that a server has already been started in `Listen`, we effectively no-op and use the existing server? And probably spit out a warning.

Reply:
Ah I see, if they call `Listen` directly. 👍🏼 Can do